src/test/java/space/obminyashka/items_exchange/service/AdvertisementServiceIntegrationTest.java

```diff
@@ -34,4 +35,22 @@ void findRandom12Thumbnails_shouldReturnEmpty() {
         final var secondTitlesGetAttempt = advertisementService.findRandom12Thumbnails();
         assertEquals(firstTitlesGetAttempt, secondTitlesGetAttempt, "Collections must be equals because of caching response");
     }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void findPaginated_shouldReturnEmpty() {
+        long id = 1;
+        int page = 0;
+        int size = 12;
+        final var firstTitlesGetAttempt = advertisementService.findByCategoryId(id, PageRequest.of(page, size));
+
+        assertAll("ValidationMessage of all parameters and mocks",
+                () -> assertFalse(firstTitlesGetAttempt.isEmpty()),
+                () -> assertEquals(repository.count(), firstTitlesGetAttempt.getTotalElements())
+        );
+
+        final var secondTitlesGetAttempt = advertisementService.findByCategoryId(1L, PageRequest.of(0, 12));
```

Comment:
I'd recommend you check how JUnit 5 parametrized tests work and update it with that kind of test.

Reply:
Deleted this test and added new parametrized test